Sources/SwiftBSON/BSON.swift

```diff
@@ -450,7 +450,11 @@ extension BSON: ExpressibleByBooleanLiteral {
 
 extension BSON: ExpressibleByDictionaryLiteral {
     public init(dictionaryLiteral elements: (String, BSON)...) {
-        self = .document(BSONDocument(dictionaryLiteral: elements))
+        do {
```

Comment:
the `ExpressibleByDictionaryLiteral` protocol does not allow us to throw from this initializer, so we must crash if the input dictionary literal is invalid